pkg/systemd/services/services.jsx

```diff
@@ -870,17 +885,35 @@ const ServicesPageFilters = ({
     fileStateDropdownOptions,
     filtersRef,
     loadingUnits,
-    onCurrentTextFilterChanged,
-    onFiltersChanged,
+    onOptionsChanged,
 }) => {
+    const { options } = usePageLocation();
+    const { activestate, filestate, name } = options;
     const [activeStateFilterIsOpen, setActiveStateFilterIsOpen] = useState(false);
-    const [currentTextFilter, setCurrentTextFilter] = useState('');
+    const [currentTextFilter, setCurrentTextFilter] = useState(decodeURIComponent(name || ""));
     const [fileStateFilterIsOpen, setFileStateFilterIsOpen] = useState(false);
     const [filters, setFilters] = useState({
-        activeState: [],
-        fileState: [],
+        activeState: JSON.parse(activestate || '[]'),
+        fileState: JSON.parse(filestate || '[]'),
     });
 
+    useEffect(() => {
+        const _filters = { activeState: JSON.parse(options.activestate || '[]'), fileState: JSON.parse(options.filestate || '[]') };
+
+        setCurrentTextFilter(decodeURIComponent(options.name || ""));
+        setFilters(_filters);
+    }, [options, setCurrentTextFilter, setFilters]);
+
+    useEffect(() => {
+        const _options = {};
+
+        _options.activestate = JSON.stringify(filters.activeState);
+        _options.filestate = JSON.stringify(filters.fileState);
```

Comment:
That JSON stringify/parse dance rings a bell :grimacing:  Still terrible, though.. (No action, just thinking aloud)

Reply:
That's for different reason here :) It's array to URL options. Thought it's enough to encode with json. WDYT?